executor/distsql.go

```diff
@@ -589,6 +593,7 @@ func (e *IndexLookUpExecutor) startIndexWorker(ctx context.Context, workCh chan<
 			SetIsStaleness(e.isStaleness).
 			SetFromSessionVars(e.ctx.GetSessionVars()).
 			SetFromInfoSchema(e.ctx.GetInfoSchema()).
+			SetClosestReplicaReadChecker(newClosestReadChecker(e.ctx, &builder.Request, e.indexNetCost/float64(len(kvRanges)))).
```

Comment:
How about making the optimizer do the decision instead of propagating net-cost to the executor?

Reply:
We can implement a new function like `useClosestReadChecker` in `plan_cost.go`, and then use it like this:
```
(p *TableReader) GetPlanCost(...) float64 {
    ...
    if useClosestReadChecker(p) {
        p.useClosestReadFloag = True
    }
    ...
}
```
And then the executor can use this flag directly.

Reply:
It will need the region information, which may also be a strange way.